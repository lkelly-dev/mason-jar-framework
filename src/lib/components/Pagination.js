import React, { useContext } from 'react';
import { cx } from 'emotion';
import { debugAttrs, debugMode } from '../utils/debugging';

const Pagination = props => {
  const { context, children, className, ...rest } = props;
  const debug = debugMode(context)
  const paginationClass = cx('card', className);
  return (
    <ul className={paginationClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </ul>
  );
};

const PaginationItem = props => {
  const { context, children, className, ...rest } = props;
  const debug = debugMode(context)
  const paginationClass = cx('card', className);
  return (
    <li className={paginationClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </li>
  );
};

const PaginationLink = props => {
  const { context, children, className, as, ...rest } = props;
  const debug = debugMode(context)
  const paginationClass = cx('page-link', className);
  const Component = as || 'a';
  return (
    <Component className={paginationClass}
      {...(debug && debugAttrs(props)) }
      {...rest}>
      {children}
    </Component>
  );
};

Pagination.Item = PaginationItem;
Pagination.Link = PaginationLink;

export { Pagination, PaginationItem, PaginationLink };
