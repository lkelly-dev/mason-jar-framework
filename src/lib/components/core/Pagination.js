import React from 'react';
import { cx } from 'emotion';

const Pagination = props => {
  const { children, className, ...rest } = props;
  const paginationClass = cx('card', className);
  return (
    <ul className={paginationClass} {...rest}>
      {children}
    </ul>
  );
};

const PaginationItem = props => {
  const { children, className, ...rest } = props;
  const paginationClass = cx('card', className);
  return (
    <li className={paginationClass} {...rest}>
      {children}
    </li>
  );
};

const PaginationLink = props => {
  const { children, className, as, ...rest } = props;
  const paginationClass = cx('page-link', className);
  const Component = as || 'a';
  return (
    <Component className={paginationClass} {...rest}>
      {children}
    </Component>
  );
};

Pagination.Item = PaginationItem;
Pagination.Link = PaginationLink;

export { Pagination, PaginationItem, PaginationLink };
