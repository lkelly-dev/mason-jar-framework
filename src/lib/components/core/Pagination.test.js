import React from 'react';
import { shallow } from 'enzyme';
import { Pagination, PaginationItem, PaginationLink } from "./Pagination";

describe('Pagination', () => {
  it('Pagination matches snapshot', () => {
    const component = shallow(<Pagination></Pagination>);
    expect(component).toMatchSnapshot();
  })
});
