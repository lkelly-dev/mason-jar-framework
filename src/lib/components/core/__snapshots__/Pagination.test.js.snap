// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Navbar matches snapshot 1`] = `
<nav
  className="navbar"
/>
`;

exports[`Pagination Pagination matches snapshot 1`] = `
<ul
  className="card"
/>
`;
